my-code/src/containers/Detail.js

```diff
@@ -0,0 +1,214 @@
+// @flow
+import React from "react"
+import { Link } from "react-router-dom"
+import omdbApi from "../constants/omdbApi"
+
+import Wrapper from "../layout/Wrapper"
+import { FlexLeft, FlexRight } from "../layout/Flex"
+
+import {
+  LargeThumbnail,
+  LargeEmptyThumbnail
+} from "../components/Thumbnail/styled"
+import Button from "../components/Button/index"
+import { IMDBLabel, RottenLabel } from "../components/Label/index"
+import Arrow from "../components/Icons/Arrow"
+
+import { Text100, Text200, Text400 } from "../components/Typography/styled"
+
+type IdType = string
+
+type PropsType = {
+  match: {
+    params: {
+      id: string
+    }
+  }
+}
+
+type StateType = {
+  info: {
+    Title?: string,
+    Runtime?: string,
+    Genre?: string,
+    Plot?: string,
+    Rated?: string,
+    Poster?: string,
+    Director?: string,
+    Actors?: string,
+    Year?: string,
+    Ratings?: {
+      Source: string,
+      Value: string
+    }
+  }
+}
+
+type ActorType = {
+  actor: string
+}
+type GenreType = {
+  actor: string
+}
+
+type DirectorType = {
+  actor: string
+}
+
+type RatingsType = {
+  Source: string,
+  Value: string
+}
+
+class Detail extends React.Component<PropsType, StateType> {
+  constructor() {
+    super()
+    this.state = {
+      info: {},
+      favourite: false,
+      active: false
```

Comment:
what's the active?
Can't we just use favorite?

That being sad, do we need the favorite key in the state or could we just use localstorage as the source-of-truth for favorites?